src/Ucommerce.Transactions.Payments.Freepay/FreepayPaymentMethodService.cs

```diff
@@ -0,0 +1,397 @@
+            {
+                status = String.Format("{0} >> {1}", PaymentMessages.CancelFailed, ex.Message);
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Acquires the payment from the payment provider. This is often used when you need to call external services to handle the acquire process.
+        /// </summary>
+        /// <param name="payment">The payment.</param>
+        /// <param name="status">The status.</param>
+        /// <returns></returns>
+        protected override bool AcquirePaymentInternal(Payment payment, out string status)
+        {
+			string apiKey = payment.PaymentMethod.DynamicProperty<string>().ApiKey.ToString();
+
+            try
+            {
+                var result = MakeRequest(JsonConvert.SerializeObject(new { Amount = payment.Amount.ToCents() }), HttpMethod.Post, API_ENDPOINT_URL + $"/api/authorization/{payment.TransactionId}/capture", apiKey, out HttpStatusCode responseCode);
+
+                if (responseCode == HttpStatusCode.OK || responseCode == HttpStatusCode.NoContent)
+                {
+                    dynamic resultData = JsonConvert.DeserializeObject(result);
+
+                    if (((JToken)resultData.IsSuccess).Value<bool>())
+                    {
+                        status = PaymentMessages.AcquireSuccess;
+                    }
+                    else
+                    {
+                        status = string.Format("{0} >> {1}", PaymentMessages.AcquireFailed, "Capture error code: " + resultData.AcquirerStatusCode);
+                    }
+                }
+                else
+                {
+                    status = string.Format("{0} >> {1}", PaymentMessages.AcquireFailed, result);
+                }
+            }
+            catch (Exception ex)
+            {
+                status = string.Format("{0} >> {1}", PaymentMessages.AcquireFailed, ex.Message);
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Refunds the payment from the payment provider. This is often used when you need to call external services to handle the refund process.
+        /// </summary>
+        /// <param name="payment">The payment.</param>
+        /// <param name="status">The status.</param>
+        /// <returns></returns>
+        protected override bool RefundPaymentInternal(Payment payment, out string status)
+        {
+			string apiKey = payment.PaymentMethod.DynamicProperty<string>().ApiKey.ToString();
+
+            try
+            {
+                var result = MakeRequest(JsonConvert.SerializeObject(new { Amount = payment.Amount.ToCents() }), HttpMethod.Post, API_ENDPOINT_URL + $"/api/authorization/{payment.TransactionId}/credit", apiKey, out HttpStatusCode responseCode);
+
+                if (responseCode == HttpStatusCode.OK || responseCode == HttpStatusCode.NoContent)
+                {
+                    dynamic resultData = JsonConvert.DeserializeObject(result);
```

Comment:
Could you find another way than using `dynamic` if something fails it is really hard to debug. We are trying to avoid using dynamic.